sn_client/src/api/file_apis.rs

```diff
@@ -518,20 +518,34 @@ mod tests {
                 tokio::spawn(async move {
                     debug!("starting client #{:?}..", i);
                     // use a fresh client
-                    let client = create_test_client().await?;
-                    debug!("client #{:?} started", i);
-                    Ok(client)
+                    let client = create_test_client().await;
+                    debug!("client #{:?} created ok?: {}", i, client.is_ok());
+                    client
                 })
                 .in_current_span();
             tasks.push(handle);
         }
 
-        join_all(tasks)
-            .await
-            .into_iter()
-            .flatten()
-            .flatten()
-            .collect()
+        let (clients, errors) = join_all(tasks).await.into_iter().flatten().fold(
+            (vec![], vec![]),
+            |mut acc, result| {
```

Comment:
Slightly easier to follow if you destructure this `acc`
```suggestion
            |(mut clients, mut errors), result| {
```